src/subprocess-posix.cc

```diff
@@ -222,6 +279,11 @@ bool SubprocessSet::DoWork() {
     ++nfds;
   }
 
+  if (nfds == 0) {
+    WaitChildExit(true);
```

Comment:
I am wondering if it would be possible to avoid handling this special case (nfds == 0) all together. I would assume that if no children have finished, DoWork() will retry and get blocked in ppoll() (with 0 fds in the set, waiting for a signal to interrupt it, including any child to finish which is what we need.
That would void the need in having two modes for WaitChildExit().